includes/Checker/Abstract_Check_Runner.php

```diff
@@ -477,16 +477,50 @@ private function get_check_exclude_slugs() {
 	 *
 	 * @since 1.0.0
 	 *
+	 * @throws Exception Throws exception if input is invalid.
+	 *
 	 * @return string The plugin basename to check.
 	 */
 	final public function get_plugin_basename() {
 		if ( null === $this->plugin_basename ) {
 			$plugin = null !== $this->plugin ? $this->plugin : $this->get_plugin_param();
 
-			if ( is_dir( $plugin ) ) {
-				$this->plugin_basename = $plugin;
-			} else {
+			// Check if a slug.
+			if ( preg_match( '/^[a-z0-9]+(-?[a-z0-9]+)*(\.php)?$/i', $plugin ) ) {
```

Comment:
A regex check is error prone. Can you elaborate on this?

Why not also just check if it contains a plugin header?

Reply:
We could have inputs like this:
```
sample-slug
hello.php
/var/www/path/plugin/
https://example.com/files/plugin-slug.zip (In the future PR)
```
This is for checking which routes should we use for finding basename.

Reply:
Now wondering how WP-CLI handles this. 😊

Reply:
CLI does not seem to accept folder as input. `wp plugin install` **plugin|zip|url** - Accepts a plugin slug, the path to a local zip file, or a URL to a remote zip file.

Reply:
Removed the regex.